frontend/src/screens/Encyclopedia.tsx

```diff
@@ -1,176 +1,56 @@
-
-  if (isNaN(currentMushroomIndex))
-    content = (
-      <View style={mushroomStyles.screen}>
-        {mushrooms.map((obj: Mushroom) => (
-          <MushroomView
-            style={mushroomStyles.mw}
-            obj={obj}
-            onChange={setScreen}
-          />
-        ))}
-      </View>
-    );
-  else
-    content = (
-      <MushroomScreen
-        obj={mushrooms[currentMushroomIndex]}
-        onChange={setScreen}
-      />
-    );
-
-  return <ScrollView>{content}</ScrollView>;
-};
-
-const MushroomScreen: React.FC<Props> = ({ obj, onChange }) => {
-  // <ScrollView contentContainerStyle={{ flexGrow: 1 }} style = {mushroomStyles.screen}>
+  const [screen, setScreen] = useState({ screen: "mushroom", mushroomId: NaN });
   return (
-    <View style={mushroomScreenStyles.screen}>
-      <Text style={mushroomScreenStyles.title}>{obj.name}</Text>
-      <View style={mushroomScreenStyles.imageContainer}>
-        <Image
-          style={mushroomScreenStyles.image}
-          source={MushList.list[obj.index].image}
-        />
-      </View>
-      <View>
-        <CategoryName text="Где можно найти:" />
-        <CategoryText text={obj.zones} />
-      </View>
-      <View>
-        <CategoryName text="Семейство:" />
-        <CategoryText text={obj.familie} />
-      </View>
-      <View>
-        <CategoryName text="Красная книга:" />
-        <CategoryText text={obj.redBook} />
-      </View>
-      <View>
-        <CategoryName text="Съедобность:" />
-        <CategoryText text={obj.eatable} />
-      </View>
-      <View>
-        <CategoryName text="Карты:" />
-        <CategoryText text={obj.gmapsLink} />
-      </View>
-      <View>
-        <CategoryName text="Описание:" />
-        <CategoryText text={obj.description} />
-      </View>
-      <TouchableOpacity
-        style={mushroomScreenStyles.closeButton}
-        onPress={() => {
-          currentMushroomIndex = NaN;
-          onChange("encyclopediaView");
-        }}
-      >
-        <Text style={mushroomScreenStyles.closeText}>Назад</Text>
-      </TouchableOpacity>
-    </View>
-  );
-};
-//
-
-const CategoryName: React.FC<Props> = ({ text }) => {
-  return <Text style={mushroomScreenStyles.categoryName}>{text}</Text>;
-};
-const CategoryText: React.FC<Props> = ({ text }) => {
-  return <Text style={mushroomScreenStyles.categoryText}>{text}</Text>;
-};
-
-const MushroomView: React.FC<Props> = ({ obj, onChange }) => {
-  return (
-    <TouchableOpacity
-      wrap
-      style={mushroomStyles.postButton}
-      onPress={() => {
-        currentMushroomIndex = obj.index;
-        onChange("encyclopedia");
-      }}
-    >
-      <View style={mushroomStyles.postContainer}>
-        <Image
-          style={mushroomStyles.image}
-          source={MushList.list[obj.index].image}
-        />
-        <View style={mushroomStyles.titleContainer}>
-          <Text style={mushroomStyles.text}>{obj.name}</Text>
+    <ScrollView>
+      {isNaN(screen.mushroomId) ? (
```

Comment:
возможно лучше сделать проверку по экрану, а не по наличию числа?
```suggestion
      {screen.screen === "encyclopediaView" ? (
```